### PR TITLE
feat(sbb-timetable-row): enhance trip status and display logic

### PR DESCRIPTION
**Pull request by WalkingOS**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/sbb-design-systems/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../docs/REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

-

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [ ] Chrome Desktop
- [ ] Edge Desktop
- [x] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [x] VoiceOver Safari Desktop
- [x] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/pearl-chain/pearl-chain.ts [hunk 1]
```diff
@@ -58,7 +62,11 @@ export class SbbPearlChainElement extends LitElement {
   }
 
   private _isAllCancelled(legs: PtRideLeg[]): boolean {
```
Comment (jeripeierSBB) [inline src/components/pearl-chain/pearl-chain.ts:64]: probably intended but sound kind of wrong that "all" canceled means that just some leg is cancelled or only even partially cancelled?

File: src/components/pearl-chain/pearl-chain.sample-data.ts [hunk 2]
```diff
@@ -4,19 +4,30 @@ const future = '2022-12-07T12:11:00+01:00';
...
+const cancelledService = { ...defaultService, serviceAlteration: { cancelled: true } };
+const delayedService = { ...defaultService, serviceAlteration: { delay: true } };
+const isNotReachableService = { ...defaultService, serviceAlteration: { reachable: false } };
+const unplannedStopService = {
+  ...defaultService,
+  serviceAlteration: { unplannedStopPointsText: 'unplannedStop' },
+};
+const redirectedService = {
+  ...defaultService,
+  serviceAlteration: { redirectedText: 'Ausnahmsweise kein Halt' },
```
Comment (jeripeierSBB) [inline src/components/pearl-chain/pearl-chain.sample-data.ts:27]: If possible, use English text, as we try to avoid any german phrases in lyne

**Comment by jeripeierSBB** [comment]
Please also check test and build run because they fail: https://github.com/sbb-design-systems/lyne-components/actions/runs/9187404639/job/25265088222?pr=2680

File: src/components/pearl-chain/pearl-chain.ts (hunk 1 above)
Comment (WalkingOS) [reply src/components/pearl-chain/pearl-chain.ts:64]: Yes, it's intended. It should appear as if it's completely canceled, including the partially canceled ones and those with canceled legs

File: src/elements-experimental/pearl-chain/pearl-chain.ts [hunk 3]
```diff
@@ -65,7 +69,11 @@ export class SbbPearlChainElement extends LitElement {
   }
 
   private _isAllCancelled(legs: PtRideLeg[]): boolean {
-    return legs?.every((leg) => leg?.serviceJourney?.serviceAlteration?.cancelled);
+    return legs?.some(
+      (leg) =>
+        leg?.serviceJourney?.serviceAlteration?.cancelled ||
+        leg?.serviceJourney?.serviceAlteration?.partiallyCancelled,
```
Comment (WalkingOS) [inline src/elements-experimental/pearl-chain/pearl-chain.ts:75]: this line should be removed :)

**Review by jeripeierSBB — APPROVED** [review]
LGTM

**Pull request merged**